flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/history/HistoryServerStaticFileServerHandler.java

```diff
@@ -166,10 +194,40 @@ private void respondWithFile(ChannelHandlerContext ctx, HttpRequest request, Str
                             }
                         }
                     }
+                    // try to unzip archived files
+                    if (enableUnzip && !success) {
+                        // extract jobid from requestPath
+                        String jobId = extractJobId(requestPath);
+                        if (!StringUtils.isNullOrWhitespaceOnly(jobId)) {
+                            // submit unzip Task and get future
+                            Boolean unzipped =
+                                    CompletableFuture.supplyAsync(
+                                                    unzipTask.apply(jobId), unzipExecutor)
+                                            .get(UNZIP_TIMEOUT, TimeUnit.SECONDS);
+                            if (unzipped && file.exists()) {
+                                success = true;
+                            }
```

Comment:
The code can be simplified to: `success = unzipped && file.exists()`

Reply:
Done